### PR TITLE
BUG: Better Cubeviz support for Spectrum1D without WCS

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request was in #1698 but is has nothing to do with loading GIF or plain cube. This is more like a pre-requisite to make it possible because it fixes existing bugs related to that feature, so I turned this into a separate PR.

Screenshot from the WFC3/IR concept notebook:

Screenshot 2022-10-12 203934

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set?
- [x] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Comment by pllim** [comment]
Rebased. Maybe we can pick this up for review next sprint?

File: jdaviz/configs/cubeviz/helper.py [hunk 1]
```diff
@@ -38,7 +38,8 @@ def _set_spectrum_x_axis(self, msg):
                         viewer.state.reset_limits()
                     break
             else:
-                viewer.state.x_att_pixel = ref_data.id["Pixel Axis 2 [x]"]
+                viewer.state.x_att = ref_data.id["Pixel Axis 2 [x]"]
+                viewer.state.reset_limits()
```
Comment (kecnry) [inline jdaviz/configs/cubeviz/helper.py:42]: why was this needed to be added?

File: jdaviz/configs/cubeviz/plugins/parsers.py [hunk 2]
```diff
@@ -57,7 +57,7 @@ def parse_data(app, file_obj, data_type=None, data_label=None):
             spectrum_viewer_reference_name=spectrum_viewer_reference_name,
             uncert_viewer_reference_name=uncert_viewer_reference_name
         )
-    elif isinstance(file_obj, str) and os.path.exists(file_obj):
+    elif isinstance(file_obj, str):
```
Comment (kecnry) [inline jdaviz/configs/cubeviz/plugins/parsers.py:60]: does this now just allow an error to be raised at the `fits.open` line if a string that is not a valid file path is passed?  Do we need to catch that case and raise a more intuitive error message?

**Review by kecnry — APPROVED** [review]
Code and tests look reasonable to me.  Just have a few follow-up questions, but they're all minor so I'll approve.  Thanks!

**Review by duytnguyendtn — APPROVED** [review]
Thanks for the fix! Works well on my end

File: jdaviz/configs/cubeviz/helper.py (hunk 1 above)
Comment (pllim) [reply jdaviz/configs/cubeviz/helper.py:42]: To be consistent with similar logic above:

https://github.com/spacetelescope/jdaviz/blob/a7837c447c9eab53ae4b65810c1e61115ab81e8c/jdaviz/configs/cubeviz/helper.py#L37-L38

File: jdaviz/configs/cubeviz/plugins/parsers.py (hunk 2 above)
Comment (pllim) [reply jdaviz/configs/cubeviz/plugins/parsers.py:60]: No, the "check if file exists" logic never reach this line. It is caught in app level here, so this check is redundant and will never be executed.

https://github.com/spacetelescope/jdaviz/blob/a7837c447c9eab53ae4b65810c1e61115ab81e8c/jdaviz/app.py#L509
